addons/edi_notifier/models/notifier.py

```diff
@@ -167,13 +178,28 @@ def action_view_cron(self):
         }
         return action
 
+    @api.multi
+    def check_disbale_edi_notifications(self):
+        """Check safety config parameter is present and return True if it's enable, otherwise False."""
+        self.ensure_one()
+        if self.safety:
+            section, _sep, key = self.safety.rpartition('.')
+            if config.get_misc(section or self._name, key):
+                _logger.info('%s %s disabled, enable by configuring safety %s',
+                             self._name, self.name,
+                             '.'.join((section or self._name, key)))
+                return True
+        return False
+
     @api.multi
     def notify(self, event_type, recs=None):
-        for notifier in self:
-            if notifier.active:
-                self.env[notifier.model_id.model].notify(
-                    notifier, event_type, recs
-                )
+        """Check for edi notification safety"""
+        if not self.check_disbale_edi_notifications():
```

Comment:
Typo: disbale --> disable

Reply:
This check should be inside the loop, otherwise this function allows self to be nothing or more than one element in the recordset, and `check_disbale_edi_notifications()`  ensures one.